core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/handlers/DeleteHandler.java

```diff
@@ -58,4 +58,24 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
   protected boolean withConfirmDeletion() {
     return true;
   }
+  
+  private List<EObject> getEObjectsToRemove(List<?> elementsToRemove) {
+    List<EObject> eObjects = elementsToRemove.stream()//
+        .filter(EObject.class::isInstance) //
+        .map(EObject.class::cast).collect(Collectors.toList());
+    
+    List<EObject> eObjectsToRemove = new ArrayList<EObject>();
+    for (EObject elem : eObjects) {
```

Comment:
I think its what does CapellaAdapterHelper.resolveDescriptorsOrBusinessObjects(eObjects)

Reply:
It should be `CapellaAdapterHelper.resolveBusinessObject`, we can not use `CapellaAdapterHelper.resolveDescriptorsOrBusinessObjects(eObjects)` since we want to keep `DRepresentationDescriptor` as they are, and not obtain their target (I guess if the user clicks on a descriptor we want to delete the descriptor, and not the semantic target).